util/misc.go

```diff
@@ -423,6 +423,12 @@ func ColumnToProto(c *model.ColumnInfo, forIndex bool) *tipb.ColumnInfo {
 	if forIndex {
 		// Use array type for read the multi-valued index.
 		pc.Tp = int32(c.FieldType.ArrayType().GetType())
+		if c.FieldType.IsArray() {
```

Comment:
If the data stored in TiKV is already utf8mb4 (data stored before this PR), is there a compatibility problem with binary when reading here?